SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Utils/Interest/ResultTypes/ResultTypesTest.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+#include "Mocks/SpatialClassInfoManagerMock.h"
+
+#include "Utils/Interest/ResultTypes.h"
+
+#define RESULT_TYPES_TEST(TestName) \
+	GDK_TEST(Core, ResultTypes, TestName)
+
+// Sanity check tests for creating result types.
+// Run tests inside the SpatialGDK namespace in order to test static functions.
+namespace SpatialGDK
+{
+	TArray<Worker_ComponentId> DataComponentIds = { 1, 2 };
+	TArray<Worker_ComponentId> OwnerOnlyComponentIds = { 3, 4 };
+	TArray<Worker_ComponentId> HandoverComponentIds = { 5, 6 };
+
+	USpatialClassInfoManagerMock* CreateAndPopulateMockClassInfoManager()
+	{
+		USpatialClassInfoManagerMock* ClassInfoManager = NewObject<USpatialClassInfoManagerMock>();
+
+		// Ensure the mock won't be garbage collected
+		ClassInfoManager->AddToRoot();
+
+		// Initialise with fake generated components
+		ClassInfoManager->SetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_Data, DataComponentIds);
+		ClassInfoManager->SetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_OwnerOnly, OwnerOnlyComponentIds);
+		ClassInfoManager->SetComponentIdsForComponentType(ESchemaComponentType::SCHEMA_Handover, HandoverComponentIds);
+
+		return ClassInfoManager;
+	}
+
+	RESULT_TYPES_TEST(GIVEN_class_info_manager_WHEN_build_client_auth_result_type_THEN_gets_correct_components)
+	{
+		USpatialClassInfoManagerMock* ClassInfoManager = CreateAndPopulateMockClassInfoManager();
```

Comment:
nit: can you add the GIVEN/WHEN/THEN comments to these tests?

Reply:
do you mean labelling where the different steps are inside them?

Reply:
Yes

Reply:
added, but it looks a bit strange